11585 rss errors

pbearne opened the pull request:
The update introduces a new feature in the SimplePie class: error caching. This will allow the SimplePie to hold errors for a specified time period before they're deleted. The changes also include a call of 'wp_feed_error_cache_transient_lifetime' filter within the new error caching function in feed.php. Error messages in widgets.php are now wrapped in an HTML "error" class for better styling control.

Trac ticket: 11585

pbearne committed:
Add error caching functionality to SimplePie class

The update introduces a new feature in the SimplePie class: error caching. This will allow the SimplePie to hold errors for a specified time period before they're deleted. The changes also include a call of 'wp_feed_error_cache_transient_lifetime' filter within the new error caching function in feed.php. Error messages in widgets.php are now wrapped in an HTML "error" class for better styling control.

pbearne committed:
Add error caching functionality to SimplePie class

The update introduces a new feature in the SimplePie class: error caching. This will allow the SimplePie to hold errors for a specified time period before they're deleted. The changes also include a call of 'wp_feed_error_cache_transient_lifetime' filter within the new error caching function in feed.php. Error messages in widgets.php are now wrapped in an HTML "error" class for better styling control.

mukeshpanchal27 commented on src/wp-includes/class-simplepie.php:
```diff
@@ -930,6 +937,16 @@ public function set_cache_duration($seconds = 3600)
+	 * Set the length of time (in seconds) that the errors for failed feed loads
+	 * will be cached
+	 *
+	 * @param int $seconds The feed error cache duration
```
Missing `@since`

mukeshpanchal27 commented on src/wp-includes/class-simplepie.php:
```diff
@@ -1645,6 +1662,11 @@ protected function fetch_data(&$cache)
 						$this->data = array();
 					}
 				}
+				// Check if the error cache has been updated
```
```suggestion
				// Check if the error cache has been updated.
```

mukeshpanchal27 commented on src/wp-includes/class-simplepie.php:
```diff
@@ -1820,6 +1841,36 @@ protected function fetch_data(&$cache)
+	/**
+	 * Cache the results of the feed request.
+	 *
+	 * @param SimplePie_Cache|false $cache Cache handler, or false to not load from the cache
```
Missing `@since`

pbearne committed:
Update src/wp-includes/class-simplepie.php

Co-authored-by: Mukesh Panchal <mukeshpanchal27@users.noreply.github.com>

pbearne committed:
Add "@since 6.5" annotation to SimplePie class methods

The changes add "@since 6.5" annotation to 'set_error_cache_duration' and 'cache the results of the feed request' methods in the SimplePie class. These annotations help to track the version introduction of these methods.

mukeshpanchal27 commented on src/wp-includes/class-simplepie.php:
```diff
@@ -561,6 +561,13 @@ class SimplePie
 	 */
 	public $cache_duration = 3600;
 
+	/**
+	* @var int Error cache duration (in seconds)
+	* @see SimplePie::set_error_cache_duration()
+	* @access private
+	*/
```
```suggestion
	/**
	 * @var int Error cache duration (in seconds)
	 * @see SimplePie::set_error_cache_duration()
	 * @access private
	 */
```

pbearne committed:
Update src/wp-includes/class-simplepie.php

Co-authored-by: Mukesh Panchal <mukeshpanchal27@users.noreply.github.com>
